usr/opt/OSC200/usr/share/man/man3/swab.3
.\" This manpage source uses rsml coding. 
.so /usr/share/lib/tmac/sml
.so /usr/share/lib/tmac/rsml
.\"
.\"
.\" *********************************************************************
.\" *                                                                   *
.\" *      Copyright (c) Digital Equipment Corporation, 1991, 1994      *
.\" *                                                                   *
.\" *                       All Rights Reserved.                        *
.\" *                                                                   *
.\" *********************************************************************
.\"
.\"
.\" HISTORY
.\"
.\" @(#)$RCSfile: swab.3,v $ $Revision: 3.2.5.2 $ (DEC) $Date: 1993/06/27 18:41:50 $
.\"
.\" (c) Copyright 1990, 1991, 1992, 1993 OPEN SOFTWARE FOUNDATION, INC. 
.\" ALL RIGHTS RESERVED 
.\" OSF/1 1.2
.\"
.\"
.TH swab 3
.SH NAME
.PP
\*Lswab\*O \- Swaps bytes
.SH LIBRARY
.PP
Standard C Library (\*Llibc.a\*O)
.\" .zA "defect, 6939, R1.1.1:  Added library line for sys5 swab."
.nL
System V Compatibility Library (\*Llibsys5.a\*O)
.\" .zZ "defect, 6939, R1.1.1:  Added library line for sys5 swab."
.SH SYNOPSIS
.PP
.sS
\*L#include <unistd.h>\*O
.PP
\*Lvoid swab(
.nL
        \*Lconst void *\*Vsrc\*L,
.nL
        \*Lvoid *\*Vdest\*L,
.nL
        \*Lsize_t \*Vnbytes\*L\|);
.sE
.\" .zA "defect, 6939, R1.1.1:  Added synopsis for sys5 swab."
.PP
System V compatibility interface:
.PP
.sS
\*L#include <unistd.h>\*O
.PP
\*Lvoid swab(
.nL
        \*Lconst char *\*Vfrom\*L,
.nL
        \*Lchar *\*Vto\*L,
.nL
        \*Lint \*Vnbytes\*L\|);
.sE
.\" .zA "defect, 6939, R1.1.1:  Added synopsis for sys5 swab."
.SH PARAMETERS
.PP
.VL
.LI "\*Vsrc\*O"
Points to the location of the string to copy.
.LI "\*Vdest\*O"
Points to the location to which the resulting string is copied.
.LI "\*Vnbytes\*O"
Specifies the number of bytes to be copied.
The \*Vnbytes\*O parameter should have an even value.
When the \*Vnbytes\*O parameter is odd, the \*Lswab(\|)\*O
function uses \*Vnbytes\*O-1 instead.
.\" .zA "defect, 6939, R1.1.1: Added that nbytes must be non-negative for sys5."
For the System V version of the \*Lswab(\|)\*O function, \*Vnbytes\*O
must also be non-negative.  If \*Vnbytes\*O is negative, nothing is
done.
.\" .zZ "defect, 6939, R1.1.1: Added that nbytes must be non-negative for sys5."
.LE
.SH DESCRIPTION
.PP
.iX "swab function"
.iX "swapping bytes"
.iX "bytes" "swapping"
The \*Lswab(\|)\*O function copies the number of bytes specified
by the \*Vnbytes\*O parameter from the location pointed
to by the \*Vsrc\*O parameter to the array pointed to by the \*Vdest\*O
parameter, exchanging adjacent bytes. If copying takes place between
objects that overlap, the behavior is undefined.
.SH NOTES
.PP
.\" .zA "defect, 6939, R1.1.1:  Added note about how to link sys5 swab."
To use the System V version of \*Lswab(\|)\*O, you must link with the
\*Llibsys5\*O library before you link with \*Llibc\*O.
.PP
.\" .zZ "defect, 6939, R1.1.1:  Added note about how to link sys5 swab."
.VL 1.5i
.LI "\*LAES Support Level:\*O"
Trial use.
.LE
.\" .ig ++
.\" .SH RETURN VALUES
.\" The \*Lswab(\|)\*O function returns no values.
.\" .++
.EQ
delim off
.EN


usr/opt/OSC200/src/usr/ccs/lib/libc/swab.c
/*
 * *****************************************************************
 * *                                                               *
 * *    Copyright (c) Digital Equipment Corporation, 1991, 1994    *
 * *                                                               *
 * *   All Rights Reserved.  Unpublished rights  reserved  under   *
 * *   the copyright laws of the United States.                    *
 * *                                                               *
 * *   The software contained on this media  is  proprietary  to   *
 * *   and  embodies  the  confidential  technology  of  Digital   *
 * *   Equipment Corporation.  Possession, use,  duplication  or   *
 * *   dissemination of the software and media is authorized only  *
 * *   pursuant to a valid written license from Digital Equipment  *
 * *   Corporation.                                                *
 * *                                                               *
 * *   RESTRICTED RIGHTS LEGEND   Use, duplication, or disclosure  *
 * *   by the U.S. Government is subject to restrictions  as  set  *
 * *   forth in Subparagraph (c)(1)(ii)  of  DFARS  252.227-7013,  *
 * *   or  in  FAR 52.227-19, as applicable.                       *
 * *                                                               *
 * *****************************************************************
 */
/*
 * HISTORY
 */
#ifndef lint
static char	*sccsid = "@(#)$RCSfile: swab.c,v $ $Revision: 4.2.7.2 $ (DEC) $Date: 1993/06/07 23:17:17 $";
#endif 
/*
 */
/*
 * (c) Copyright 1990, OPEN SOFTWARE FOUNDATION, INC.
 * ALL RIGHTS RESERVED
 */
/*
 * OSF/1 Release 1.0
 */
#if !defined(lint) && !defined(_NOIDENT)

#endif
/*
 * COMPONENT_NAME: (LIBCGEN) Standard C Library General Functions
 *
 * FUNCTIONS: swab
 *
 * ORIGINS: 3 27
 *
 * This module contains IBM CONFIDENTIAL code. -- (IBM
 * Confidential Restricted when combined with the aggregated
 * modules for this product)
 * OBJECT CODE ONLY SOURCE MATERIALS
 * (C) COPYRIGHT International Business Machines Corp. 1985, 1989 
 * All Rights Reserved
 *
 * US Government Users Restricted Rights - Use, duplication or
 * disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 *
 * Copyright (c) 1984 AT&T	
 * All Rights Reserved  
 *
 * THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF AT&T	
 * The copyright notice above does not evidence any   
 * actual or intended publication of such source code.
 *
 * swab.c	1.6  com/lib/c/gen,3.1,8943 9/9/89 09:32:44
 */

/* name space pollution clean up */
#ifdef _NAME_SPACE_WEAK_STRONG
#include "pollution.h"
#if !defined(_THREAD_SAFE)
#pragma weak swab = __swab
#endif
#endif
#include <unistd.h>
#include <limits.h>

/*
 * NAME:	swab
 *
 * FUNCTION:	swab - swap bytes in 16-bit [half-]words
 *
 * NOTES:	Swab copies 'ntypes' bytes from 'from' to
 *		'to' exchanging adjacent even and odd bytes.
 *
 * RETURN VALUE DESCRIPTION:	none
 */

void
swab(From, To, nbytes)
const void *From;			/* source			*/
void *To;			/* destination			*/
ssize_t nbytes;			/* number of bytes to copy	*/
{
       register int nshorts = nbytes/2; /* number of exchanges to make */
        register unsigned char *from = (unsigned char *)From;
        register unsigned char *to = (unsigned char *)To;

        while (nshorts-- > 0) {
                *to++ = from[1];
                *to++ = *from++;
                from++;
	}
}
